chore(clerk-js): Replace Avatar preview in OrganizationProfileAvatarUploader with Upload Button

nikospapcom opened the pull request:
## Description
Replacing AvatarPreview in `OrganizationProfileAvatarUploader` with upload button in `CreateOrganizationForm` to avoid layout shift 

<!-- 
  Thanks for contributing to Clerk. Make sure to read the contributing guide at https://github.com/clerkinc/javascript/blob/main/docs/CONTRIBUTING.md before opening a PR!

  **Please create a feature request before starting work on any significant change.**

  Write a brief description of the changes introduced in this PR.
  Include screenshots/videos if they help convey the change.

  Also explain how one can test the change.
-->

<!-- Fixes #(issue number) -->

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

BEFORE
https://github.com/clerkinc/javascript/assets/14893769/35606a68-1d68-42de-8f5b-05e4db102044

AFTER
https://github.com/clerkinc/javascript/assets/14893769/1656a9f8-6153-4d2d-ade5-034beaed40ed

![Screenshot 2023-11-02 at 11 13 13 AM](https://github.com/clerkinc/javascript/assets/14893769/6ce0c78c-874a-4d77-975d-17426470c68f)

panteliselef commented on packages/clerk-js/src/ui/components/CreateOrganization/CreateOrganizationForm.tsx:
```diff
@@ -121,6 +123,31 @@ export const CreateOrganizationForm = (props: CreateOrganizationFormProps) => {
+                    opacity: 0.8,
+                  },
+                })}
+                onMouseEnter={() => setIsHovered(true)}
+                onMouseLeave={() => setIsHovered(false)}
```
Nice touch,

What i would do is to only keep one of them (pref. the Upload icon) and just have an "zoom in" animation on hover.


```tsx
<Icon
  size='md'
  icon={Upload}
  sx={t => ({
     ':hover': {
        transform: 'scale(1.2)',
     },
  })}
/>
```

panteliselef commented on packages/clerk-js/src/ui/components/CreateOrganization/CreateOrganizationForm.tsx:
```diff
@@ -121,6 +123,31 @@ export const CreateOrganizationForm = (props: CreateOrganizationFormProps) => {
+                  borderRadius: theme.radii.$md,
+                  backgroundColor: theme.colors.$avatarBackground,
+                  ':hover': {
+                    backgroundColor: theme.colors.$avatarBackground,
+                    opacity: 0.8,
```
🔧
```suggestion
    backgroundColor: colors.makeTransparent(theme.colors.$avatarBackground, 0.2),
```

panteliselef commented on packages/clerk-js/src/ui/components/CreateOrganization/CreateOrganizationForm.tsx:
```diff
@@ -121,6 +123,31 @@ export const CreateOrganizationForm = (props: CreateOrganizationFormProps) => {
+              <IconButton
+                variant='ghost'
+                colorScheme='neutral'
+                aria-label='Upload image'
```
```suggestion
                aria-label='Upload organization logo'
```

panteliselef reviewed (approved):
Great stuff, i added a few minor comments

nikospapcom commented:
> Great stuff, i added a few minor comments

Thanks for the review @panteliselef 
I addressed you comments and update after screenshot/video accordingly

panteliselef reviewed (approved):
💯
